File: week1_programming_challenges/2_maximum_pairwise_product/max_pairwise_product.py
# python3


def max_pairwise_product(numbers):
    n = len(numbers)
    max1=0
    max2=0
    for first in range(n):
        if(numbers[first]>max1):
            max1=numbers[first]
            index=first
        
    for second in range(n):
        if(numbers[second]>max2 and second!=index):
            max2=numbers[second]
                
    max_product=max1*max2

    return max_product


if __name__ == '__main__':
    input_n = int(input())
    input_numbers = [int(x) for x in input().split()]
    print(max_pairwise_product(input_numbers))

File: week1_programming_challenges/2_maximum_pairwise_product/max_pairwise_product.cpp
#include <iostream>
#include <vector>
#include <algorithm>

long long MaxPairwiseProduct(const std::vector<long long>& numbers) {
    long long max_product = 0;
    long long n = numbers.size();

    for (long long first = 0; first < n; ++first) {
        for (long long second = first + 1; second < n; ++second) {
            max_product = std::max(max_product,
                numbers[first] * numbers[second]);
        }
    }

    return max_product;
}

int main() {
    long long n;
    std::cin >> n;
    std::vector<long long> numbers(n);
    for (long long i = 0; i < n; ++i) {
        std::cin >> numbers[i];
    }

    std::cout << MaxPairwiseProduct(numbers) << "\n";
    return 0;
}
